src/iwant_bot/start.py

```diff
@@ -24,64 +51,336 @@ def format_message(message):
+    """Create message with buttons to adjust request duration"""
+    time_minutes = round((body['duration_ts'] - body['incoming_ts']) / 60)
+    text = f"{', '.join(body['activity'])}! I am looking for someone for {time_minutes} minutes."
+    attachment = [
+        {
+            'text': 'Would you like to change the duration to:',
+            'callback_id': f"{body['callback_id']}",
+            'fallback': 'Never mind.',
+            #'color' : '#3AA3E3',
+            'attachment_type': 'default',
+            'actions': [
+                {
+                    'name': 'Time',
+                    'text': '30 min',
+                    'type': 'button',
+                    'value': '1800'
+                },
+                {
+                    'name': 'Time',
+                    'text': '1 hour',
+                    'type': 'button',
+                    'value': '3600'
+                },
+                {
+                    'name': 'Time',
+                    'text': 'Cancel',
+                    'type': 'button',
+                    'value': '0'
+                }
+            ]
+        }
+    ]
+    return {'text': text, 'attachments': attachment}
+
+
 
-    if trigger_word == 'repeat':
-        message = format_response(f'{user_name} wants me to {text}')
-    elif trigger_word == 'whoami':
-        message = format_response(f'You are {user_name} with id {user_id}')
+def iwant_activity(body: dict) -> bool:
+    """Search the text for activities.
+    Suppose a lot of assumptions...
+    ...nobody with the name @coffee, activity 'invite' etc."""
+    any_activity = False
+    body['activity'] = []
+    body['callback_id'] = []
+    for activity in _iwant_activities:
+        if re.search(activity, body['text'].lower()):
+            any_activity = True
+            body['duration_ts'] = body['incoming_ts'] + iwant_duration(body['text'])
+            body['activity'].append(activity)
+            body['callback_id'].append(create_iwant_request(body))
+            iwant_invite(body)
+    return any_activity
+
+
+def iwant_duration(text: str) -> float:
+    """Remove commands, activities, and invited people.
+    Only text with time should leave (and some preposition).
+    :return: duration of activity in seconds."""
+    activities = '|'.join(_iwant_activities)    # Variable is used in pattern.
+    text_time = re.sub(f'({activities}|@[a-z0-9][-_.a-z0-9]{{1,20}}|with|or|and|[.!?]\s*$)', '', text)
+    print(text_time)
+    now_plus_duration = parse(text_time)  #zlobi... odmazat .,; a jine symboly, napsat testy.
+    print(now_plus_duration)
+    if now_plus_duration is None:
+        return _duration  # Implicit value
     else:
-        message = format_response(f'{user_name} wrote {text}')
+        # minimal valid time is 1s and maximal is _duration_max seconds
+        return min(max((now_plus_duration - datetime.now()).total_seconds(), 1.0), _duration_max)
+
+
+def iwant_invite(body: dict):
```

Comment:
This one also does two things (parses string + reacts on result of the parsing).